src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -174,10 +174,6 @@ private void OnDiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
 
             private void OnDiagnosticsUpdated(DiagnosticsUpdatedArgs e)
             {
-                // First see if this is a document/project removal.  If so, clear out any state we
-                // have associated with any analyzers we have for that document/project.
-                ProcessRemovedDiagnostics(e);
```

Comment:
>  id is only unique per workspace

So this seems confusing - if the id is not unique across different workspaces then we need to make it explicit that it is IdPerWorksapce. Otherwise, the consumer can easily end up messing any of its state that is keyed by just the id (for example, _idToProviderAndTagger map for the AggregatingTagger) if it doesn't check the workspace that the event belongs to. Any reason we can't make id unique across all workspaces or at least ensure they can never be same for diagnostic events from workspaces of different kinds (preview, host, misc)?